src/jobs/types.ts

```diff
@@ -0,0 +1,14 @@
+export interface ReportItem {
+  count: number;
+  ids: string[];
+  errors: string[];
+  errorCount: number;
+}
```

Comment:
I really like this idea, a standardized summary for similar jobs. I would make a few recommendations for a job report:
1. In a Job report, lets add some job details like: `jobName`, `startedAt`, `finishedAt`, `success`=(true/false), `error`=(optional fatal error that crashed the job), `details`=contents of this report item
2. `ids` should be all notifications we identified as needing this notification, not just the ones that resolved successfully
3. `errors` should be an object with details such as: `id` of object that had an issue, and then details about what failed (error message should be enough). note that we could have error failures from the email send or just from saving this model to the DB.

Reply:
I like this idea! I'm a little confused about what you're talking about in `2)` though?
And for `3)`, i was trying to  create a standard error object that would have this info but I ran into issues getting all the info i needed (esp `id`) everywhere that i wanted to throw errors...

Reply:
re: point 2. - In your utilty method that builds the ReportItem, the value for `ids` only reads from teh resolved (ie. successful) email results, so you dont get the IDs of the applications that you udpated the state to paused but that failed to send notifications

Reply:
re: point 3 - You definitely want to capture the IDs of the applications that didnt send so that they can be followed up on as required. Let me know if you want a hand finding a way to capture this info.